code.js

```diff
@@ -1,3 +1,83 @@
-function quicksort(array) {
-    return array;
+// Kane Kriz
+// UWYO COSC 3020 Algorithms
+// Quicksort Exercise
+// 25 March 2025
+//
+//
+
+
+//
+
+
+function quicksort(array) //iterative as specified, no recursion
+{
+   // var pivot; //probably useless to declare this here, may have to live within the helper
+    var arrLen = array.length;
+
+   if(arrLen == 0 || arrLen == 1)
+    {
+        return array; //0 or 1 elements, no need to sort, just return
+    }
+
+    var startPos = 0;
+    var endPos = arrLen - 1;
+    var pivotIndex = partitionArr(array, startPos, endPos);
+
+    while(true)
+    {
+        if(startPos < pivotIndex - 1)
+        {
+            pivotIndex = partitionArr(array, startPos, pivotIndex - 1);
+        }
+           
+        else if(endPos > pivotIndex + 1)
+        {
+            startPos = pivotIndex + 1;
+            pivotIndex = partitionArr(array, startPos, endPos);
+        }
+           
+        else
+        {
+            if(startPos >= endPos - 1)
+            {
+               break; //exit while true via break once all partitions are processed
+            }
+            startPos = pivotIndex + 1;
```

Comment:
Not quite following you here -- this is the same code as in the condition above. Do you need two separate conditions to run the same code?